### PR TITLE
[Darwin] Add support for multiple paths for generic read/subscribe methods

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

The current generic APIs for darwin does not allow multiple paths.

#### Change overview
 * Add support for multiple paths by using `id` instead of `NSNumber *`

#### Testing
It was manually tested using `./out/debug/standalone/chip-tool-darwin any subscribe-by-id 6,6 0,0 1 5 1 0x12344321 1,2` and `./out/debug/standalone/chip-tool-darwin any read-by-id 6,6 0,0 1 5 1 0x12344321 1,2`

### CODE REVIEW

File: examples/chip-tool-darwin/commands/clusters/ModelCommandBridge.h [hunk 1]
```diff
@@ -31,16 +31,16 @@ class ModelCommand : public CHIPCommandBridge
     void AddArguments()
     {
         AddArgument("node-id", 0, UINT64_MAX, &mNodeId);
-        AddArgument("endpoint-id", 0, UINT16_MAX, &mEndPointId);
+        AddArgument("endpoint-id", 0, UINT16_MAX, &mEndPointIds);
```
Comment (isiu-apple) [inline examples/chip-tool-darwin/commands/clusters/ModelCommandBridge.h:34]: Is `AddArguments()` here coming from `examples/chip-tool/commands/common/Command.h`?  If so, shall we add the optional `desc` description string to the arguments we are modifying in this review?

File: src/darwin/Framework/CHIP/CHIPDevice.h [hunk 2]
```diff
@@ -175,9 +175,9 @@ extern NSString * const kCHIPArrayValueType;
 /**
  * Read attribute in a designated attribute path
  */
-- (void)readAttributeWithEndpointId:(NSNumber * _Nullable)endpointId
-                          clusterId:(NSNumber * _Nullable)clusterId
-                        attributeId:(NSNumber * _Nullable)attributeId
+- (void)readAttributeWithEndpointId:(id _Nullable)endpointId
+                          clusterId:(id _Nullable)clusterId
+                        attributeId:(id _Nullable)attributeId
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/CHIPDevice.h:180]: So now that I am looking at this API... this is really pretty annoying to use: you have to create three lists that line up, etc.

I think we should have a public `readAttributePaths` API, alongside the existing `readAttributeWithEndpointId`, which can call `readAttributePaths` internally.  And readAttributePaths should take a single list of paths, where each path is an object with three nullable members for endpoint/cluster/attribute, null representing wildcard.

File: src/darwin/Framework/CHIP/CHIPDevice.mm [hunk 3]
```diff
@@ -1182,23 +1226,29 @@ - (void)subscribeAttributeWithEndpointId:(NSNumber * _Nullable)endpointId
...
-        if (endpointId) {
-            container.pathParams->mEndpointId = static_cast<chip::EndpointId>([endpointId unsignedShortValue]);
-        }
-        if (clusterId) {
-            container.pathParams->mClusterId = static_cast<chip::ClusterId>([clusterId unsignedLongValue]);
-        }
-        if (attributeId) {
-            container.pathParams->mAttributeId = static_cast<chip::AttributeId>([attributeId unsignedLongValue]);
+
+        chip::app::AttributePathParams attributePathParams[kMaxAllowedPaths];
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/CHIPDevice.mm:1230]: But then the ReadClient will have a pointer to this stack object, and if it tries to resubscribe things will blow up, no?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Comment by vivien-apple** [comment]
Closing until I got some time to work on that again.

**Pull request closed without merging**